common/common.cpp

```diff
@@ -1188,6 +1188,7 @@ void dump_string_yaml_multiline(FILE * stream, const char * prop_name, const cha
     if (!data_str.empty() && (std::isspace(data_str[0]) || std::isspace(data_str.back()))) {
         data_str = std::regex_replace(data_str, std::regex("\n"), "\\n");
         data_str = std::regex_replace(data_str, std::regex("\""), "\\\"");
+        data_str = std::regex_replace(data_str, std::regex("\\\\[^n\"]"), "\\$&");
```

Comment:
I think raw string literals would be clearer here. And I think it should be ordered like this so a `\n` in the input is correctly escaped as `\\n` (correct me if I'm wrong):
```suggestion
        data_str = std::regex_replace(data_str, std::regex(R"(\\)"), R"(\\)");
        data_str = std::regex_replace(data_str, std::regex(R"(\n)"), R"(\n)");
        data_str = std::regex_replace(data_str, std::regex(R"(")"), R"(\")");
```

Reply:
Thank you for the feedback :-)

I see your suggestion regarding the raw string :-) I agree it makes it less confusing, but on the other side, there is also some mind gymnastics involved in understanding why we need to do the:
```
data_str = std::regex_replace(data_str, std::regex(R"(\n)"), R"(\n)");
```
 replacing the same? (I mean, it is because the regex string needs the escaping for that matching, but is that obvious for the casual code reader 😄 )

I included your proposal to use raw strings, and thank you for the feedback.